C-Plus-Plus/cp/infix_to_postfix.cpp

```diff
@@ -0,0 +1,176 @@
+		else if(IsOperand(expression[i]))
+		{
+			postfix +=expression[i];
+		}
+
+		//If character is operator, check for Higher precedence operator in Stack top
+		//till first opening bracket and pop all such operators
+		//Finally push the current operator on the Stack
+		else if(IsOperator(expression[i]))
+		{
+			while(!S.empty() && !IsOpeningParentheses(S.top())
+            		&& HasHigherPrecedence(S.top(),expression[i]))
+			{
+				postfix+= S.top();
+				S.pop();
+			}
+			S.push(expression[i]);
+		}
+
+		//If opening Parentheses simply push it on the Stack
+		else if(IsOpeningParentheses(expression[i]))
+		{
+			S.push(expression[i]);
+		}
+
+        //If Closing Parentheses then pop all the operators and append to res string
+        // until Stack top is opening Parentheses and finally one extra pop for Opening Parentheses
+		else if(IsClosingParentheses(expression[i]))
+		{
+			while(!S.empty() && !IsOpeningParentheses(S.top())) {
+				postfix += S.top();
+				S.pop();
+			}
+			S.pop();
+		}
+	}
+
+    //Finally pop and append all operators till Stack is not empty
+	while(!S.empty()) {
+		postfix += S.top();
+		S.pop();
+	}
+
+	return postfix;
+}
+
+// Function to verify whether a character is english letter or numeric digit.
+// We are assuming in this solution that operand will be a single character
+bool IsOperand(char C)
+{
+	if(C >= '0' && C <= '9') return true;
+	if(C >= 'a' && C <= 'z') return true;
+	if(C >= 'A' && C <= 'Z') return true;
+	return false;
+}
+
+// Function to verify whether a character is operator symbol or not.
+bool IsOperator(char C)
+{
+	if(C == '+' || C == '-' || C == '*' || C == '/' || C== '^')
+		return true;
+
+	return false;
+}
+
+bool IsOpeningParentheses(char C)
+{
+    if(C == '(' || C == '{' || C=='[')
+        return true ;
+    return false;
+}
+
+bool IsClosingParentheses(char C)
+{
+    if(C == ')' || C == '}' || C==']')
+        return true ;
+    return false;
+}
+// Function to verify whether an operator is right associative or not.
+int IsRightAssociative(char op)
+{
+	if(op == '^') return true;
+	return false;
+}
+
+// Function to get weight of an operator. An operator with higher weight will have higher precedence.
+int GetOperatorWeight(char op)
+{
+	int weight = -1;
+	switch(op)
+	{
+	case '+':
+	case '-':
+		weight = 1;
+		break ;
+	case '*':
+	case '/':
+		weight = 2;
+		break ;
+	case '^':
+		weight = 3;
+		break ;
+	}
+	return weight;
+}
+
+// Function to perform an operation and return output.
+int HasHigherPrecedence(char op1, char op2)
+{
+	int op1Weight = GetOperatorWeight(op1);
+	int op2Weight = GetOperatorWeight(op2);
+
+	// If operators have equal precedence, return true if they are left associative.
+	// return false, if right associative.
+	// if operator is left-associative, left one should be given priority.
+	if(op1Weight == op2Weight)
+	{
+		if(IsRightAssociative(op1)) return false;
+		else return true;
+	}
+	return op1Weight > op2Weight ?  true: false;
+}
```

Comment:
Add sample input/output in commented form.

Reply:
@atarax665 i added the sample input and its output.